clients/include/testing_scal_strided_batched.hpp

```diff
@@ -0,0 +1,238 @@
+    rocblas_int incx        = arg.incx;
+    rocblas_int stridex     = arg.stride_x;
+    rocblas_int batch_count = arg.batch_count;
+    U           h_alpha     = arg.get_alpha<U>();
+
+    rocblas_local_handle handle;
+
+    // argument sanity check before allocating invalid memory
+    if(stridex < N * incx)
+    {
+        static const size_t safe_size = 100; // arbitrarily set to 100
+        device_vector<T>    dx(safe_size);
+        if(!dx)
+        {
+            CHECK_HIP_ERROR(hipErrorOutOfMemory);
+            return;
+        }
+
+        CHECK_ROCBLAS_ERROR(rocblas_set_pointer_mode(handle, rocblas_pointer_mode_host));
+        EXPECT_ROCBLAS_STATUS((rocblas_scal_strided_batched<T, U>)(handle,
+                                                                   N,
+                                                                   &h_alpha,
+                                                                   dx,
+                                                                   incx,
+                                                                   stridex,
+                                                                   batch_count),
+                              rocblas_status_invalid_size);
+        return;
+    }
+    if(N <= 0 || incx <= 0 || batch_count <= 0)
+    {
+        static const size_t safe_size = 100; // arbitrarily set to 100
+        device_vector<T>    dx(safe_size);
+        if(!dx)
+        {
+            CHECK_HIP_ERROR(hipErrorOutOfMemory);
+            return;
+        }
+
+        CHECK_ROCBLAS_ERROR(rocblas_set_pointer_mode(handle, rocblas_pointer_mode_host));
+        if(batch_count < 0)
+            EXPECT_ROCBLAS_STATUS((rocblas_scal_strided_batched<T, U>)(handle,
+                                                                       N,
+                                                                       &h_alpha,
+                                                                       dx,
+                                                                       incx,
+                                                                       stridex,
+                                                                       batch_count),
+                                  rocblas_status_invalid_size);
+        else
+            CHECK_ROCBLAS_ERROR((rocblas_scal_strided_batched<T, U>)(handle,
+                                                                     N,
+                                                                     &h_alpha,
+                                                                     dx,
+                                                                     incx,
+                                                                     stridex,
+                                                                     batch_count));
+
+        return;
+    }
+
+    size_t size_x = N * size_t(incx) * batch_count
+                    + (static_cast<size_t>(stridex) - N * size_t(incx))
```

Comment:
size_t( ) instead of static_cast , maybe we should comment on the stride assumptions when calculating